examples/vision/semisupervised_simclr.py

```diff
@@ -0,0 +1,631 @@
+"""
+Title: Semi-supervised image classification using contrastive pretraining with SimCLR
+Author: [András Béres](www.linkedin.com/in/andras-beres-789190210)
+Date created: 2021/04/24
+Last modified: 2021/04/24
+Description: Contrastive pretraining with SimCLR for semi-supervised image classification on the STL-10 dataset.
+"""
+"""
+## Introduction
+
+### Semi-supervised learning
+
+Semi-supervised learning is a machine learning paradigm, that deals with **partially
+labeled datasets**. When applying deep learning in the real world, one usually has to
+gather a large dataset to make it work well. However, while the cost of labeling scales
+linearly with the dataset size (labeling each example takes a constant time), model
+performance only scales [sublinearly](https://arxiv.org/abs/2001.08361) with it. This
+means that labeling more and more samples becomes less and less cost-efficient, while
+gathering unlabeled data is generally cheap as it is usually readily available in large
+quantities.
+
+Semi-supervised learning offers to solve this problem by only requiring a partially
+labeled dataset, and by being label-efficient by utilizing the unlabeled examples for
+learning as well.
```

Comment:
I'm just reviewing the text and noticed something - are you wrapping at a certain character level? 

E.g. At `80` this paragraph would look as follows:

```suggestion
Semi-supervised learning offers to solve this problem by only requiring a
partially labeled dataset, and by being label-efficient by utilizing the
unlabeled examples for learning as well.
```

(I'm reviewing the syntax and grammar and, for example, this sentence has "...by ... by ... by" structure - I will send your the suggestions soon.)

Reply:
Yeah, they weren't wrapped properly after I edited the text manually, its fixed now.